README.md

```diff
@@ -2,24 +2,75 @@
 
 Because everybody wants a pony.
 
+Pony serves gRPC methods by marshaling the method request proto to json and
+passing that to a [jsonnet] function. The result of the jsonnet function is
+unmarshaled into the response proto for return to the caller.
+
+This allows quick and dirty mocking/faking/stubbing of gRPC services in a fairly
+dynamic way, making it easy to tweak values to explore how a client interacts
+with a service.
+
+[jsonnet]: https://jsonnet.org
+
+
+## Configuration
+
+Generate a binary protobuf definition of your service to stub, including
+transitive dependencies:
+
+    protoc -o service.pb --include_imports service.proto
+
+(add other `protoc` flags and arguments as required)
+
+Put jsonnet method definitions together in a directory, each named named
+`<pkg>.<service>.<method>.jsonnet`. Each jsonnet file is evaluated when the gRPC
+server receives a call to that method.
+
+The request protobuf message is marshaled to JSON and set as the `input` jsonnet
+top-level argument when evaluating the jsonnet method. It is marshaled as:
+
+    {
+        request: { ...json-encoded protobuf here... }
+    }
+
+The jsonnet method definition should be a function of the following form:
+
+    function(input) {
+        response: { ...json-encoded protobuf here... }
+    }
+
+The response can reference fields of the input using regular jsonnet references.
+See the [testdata samples](./testdata) for an example.
+
+The `request` and `response` fields are encoded from/to protobuf messages
+according to the [protojson] encoding rules.
+
+To serve these jsonnet methods, run:
+
+    pony serve --proto-set=service.pb --method-dir=dir
+
+
+[protojson]: https://developers.google.com/protocol-buffers/docs/proto3#json
+
+
 ## Playing
 
-Build and start the echo gRPC server with
+Build and start pony on the test data:
 
-	. ./bin/activate-hermit
-	make install
-	server
+    . ./bin/activate-hermit
+    make install pb
+    pony serve --proto-set testdata/all.pb --method-dir testdata
 
 in a second terminal call it with
 
-	client
+    client world
 
-Instead of `server` try starting `dynamicserver` to run a gRPC server
-without code generation.
+Experiment with the jsonnet method file in the [testdata](./testdata)
+directory.
```

Comment:
Alternatively there is a "real" gRPC server that the same `client` can interact with. Start it with `server`.

Reply:
$ client Neo
🏢: Run Neo, run

Reply:
done